### PR TITLE
refactor: human-readable resource IDs use double underscore

### PR DESCRIPTION
**Pull request by yonadaaa**

### CODE REVIEW

File: .changeset/dirty-lizards-tell.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+"@latticexyz/store-sync": patch
+"@latticexyz/dev-tools": patch
+"@latticexyz/common": patch
+"@latticexyz/cli": patch
+---
+
+Updated all human-readable Resource ID's to use `{namespace}__{name}`.
```
Comment (holic) [inline .changeset/dirty-lizards-tell.md:8]:
```suggestion
Updated all human-readable resource ID's to use `{namespace}__{name}` for consistency with world function selectors.
```

File: packages/common/src/resourceLabel.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export type ResourceLabel = `${string}__${string}`;
+
+export function resourceLabel({
```
Comment (holic) [inline packages/common/src/resourceLabel.ts:3]: thoughts on `getResourceLabel` or `resourceToLabel`?

File: packages/common/src/resourceLabel.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export type ResourceLabel = `${string}__${string}`;
```
Comment (holic) [inline packages/common/src/resourceLabel.ts:1]: if you wanted to get real fancy, could make this generic, e.g.

```ts
export type ResourceLabel<namespace extends string = string, name extends string = string> = `${namespace}__${name}`;
```

**Review by holic — DISMISSED** [review]
just some tiny stylistic things, nothing particularly blocking

File: packages/common/src/resourceLabel.ts (hunk 3 above)
Comment (yonadaaa) [reply packages/common/src/resourceLabel.ts:1]: Ohhh I like that! I was thinking `namespace` and `name` should kinda be types

File: packages/common/src/resourceLabel.ts (hunk 2 above)
Comment (yonadaaa) [reply packages/common/src/resourceLabel.ts:3]: Definitely prefer one of them

File: .changeset/dirty-lizards-tell.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+"@latticexyz/store-sync": patch
+"@latticexyz/dev-tools": patch
+"@latticexyz/common": patch
+"@latticexyz/cli": patch
+---
+
+Updated all human-readable resource ID's to use `{namespace}__{name}` for consistency with world function signatures.
```
Comment (holic) [inline .changeset/dirty-lizards-tell.md:8]:
```suggestion
Updated all human-readable resource IDs to use `{namespace}__{name}` for consistency with world function signatures.
```

File: packages/common/src/resourceLabel.ts (hunk 2 above)
Comment (yonadaaa) [reply packages/common/src/resourceLabel.ts:3]: Going with `resourceToLabel` for consistency with `resourceToHex`

File: packages/common/src/resourceLabel.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+export function resourceLabel({
+  namespace,
+  name,
+}: {
+  readonly namespace: string;
+  readonly name: string;
```
[the remark this answers is not in the document]
Comment (yonadaaa) [reply packages/common/src/resourceLabel.ts]: Kicking this one until we fix `Resource`: https://github.com/latticexyz/mud/issues/2321

**Pull request merged**